tools/autograd/gen_view_funcs.py

```diff
@@ -0,0 +1,315 @@
+# Generates ViewFuncs.h/cpp
+#
+# NOTE: If any changes are being made to the ViewFunc codegen please also check
+# if updates are needed in torch/csrc/autograd/autograd_not_implemented_fallback.cpp
+# The fallback is expected to mimic this codegen, so we should keep the two in sync.
+
+from typing import List, Tuple
+
+import torchgen.api.dispatcher as dispatcher
+from torchgen.api.autograd import NativeFunctionWithDifferentiabilityInfo
+from torchgen.api.translate import translate
+from torchgen.api.types import (
+    BaseCType,
+    Binding,
+    NamedCType,
+    SymIntT,
+    tensorT,
+    VectorCType,
+)
+from torchgen.code_template import CodeTemplate
+from torchgen.model import Argument, NativeFunction, OptionalType
+from torchgen.utils import FileManager
+
+from .gen_inplace_or_view_type import (
+    CALL_DISPATCH,
+    extract_bindings,
+    get_view_info,
+    modifies_arguments,
+    use_derived,
+)
+
+FUNCTION_DECLARATION = CodeTemplate(
+    """\
+#define ${uppercase_op}_AVAILABLE
+#ifdef _WIN32
```

Comment:
So, what exactly is going on with the WIN32 ifdef here?

Reply:
Did you run out of exportable symbols? But if Windows doesn't need this, then why does Linux need these classes exported?

Reply:
tbh I cargo culted this from autograd's `Functions.h` :p I'll remove it and see if anything breaks